compiler/rustc_passes/src/loops.rs

```diff
@@ -230,11 +280,31 @@ impl<'a, 'hir> CheckLoopVisitor<'a, 'hir> {
                 self.sess.dcx().emit_err(BreakInsideAsyncBlock { span, closure_span, name });
             }
             UnlabeledBlock(block_span) if is_break && block_span.eq_ctxt(break_span) => {
-                let suggestion = Some(OutsideLoopSuggestion { block_span, break_span });
-                self.sess.dcx().emit_err(OutsideLoop { span, name, is_break, suggestion });
+                if let Some(block) = self.block_breaks.get_mut(&block_span) {
+                    block.spans.push(span);
+                    block.suggs.push(break_span);
+                } else {
+                    self.blocks.push(block_span);
```

Comment:
Do we *need* `self.blocks` to be around? Can't we iterate in `block_breaks` down in `report_outside_loop_error`?

Reply:
> Do we _need_ `self.blocks` to be around? Can't we iterate in `block_breaks` down in `report_outside_loop_error`?

Thank you. Because iterate in `block_breaks`(`HashMap`) is disordered. So I add `blocks`. `blocks` only used for this, should I remove it?